habitat_baselines/evaluate_ppo.py

```diff
@@ -149,13 +219,82 @@ def main():
         episode_counts += 1 - not_done_masks
         current_episode_reward *= not_done_masks
 
+        # added for video support
+        next_episodes = envs.current_episodes()
+        envs_to_pause = []
+        n_envs = envs.num_envs
+        for i in range(n_envs):
+            if next_episodes[i].episode_id in stats_episodes:
+                envs_to_pause.append(i)
+
+            if not_done_masks[i].item() == 0:
+                # episode ended, record and generate videos as specified
+                stats_episodes.add(current_episodes[i].episode_id)
+                if args.video_option != 0 and len(rgb_frames[i]) > 0:
```

Comment:
Abstract out the video generation logic into a separate method, the main function is becoming really cumbersome here.